src/verifier.js

```diff
@@ -63,10 +66,12 @@ class LocalVerifier {
   verify(req, username, password, done) {
     debug('Checking credentials', username, password);
     const query = {
-      [this.options.usernameField]: username,
       $limit: 1
     };
 
+    // set usernameField in query, accepting dot notation
+    set(query, this.options.usernameField, username);
```

Comment:
Hm, I see. I have no issue with writing custom verifier for applications; that makes sense because every application has different needs and requirements.

I was originally thinking that the user would use dot notation for `usernameField` if he knew his database supported it, and if it doesn't then just write the root field. Though, that still doesn't solve the problem of MongoDB's not using actual nested objects.

And then of course there's the error handling (letting the user know that the usernameField he specified is incorrect for his specific database, as opposed to no user being found).

It could get messy pretty quickly, so a custom verifier sounds like the way to go.

I'm not sure if this is being done yet, but maybe in the future we could attach some data to the service that specifies which type of database driver the user has attached to the service. That would allow us to perform different operations inside of feathers packages based on those values. I'd say that's a bit further in the future (and if that were to be done, things would likely also have to get a bit more strict in terms of which drivers are allowed into feathers, which kind of goes against the feathers ideology of being lightweight and customizable).

Something to think about for sure.